### DISCUSSION TITLE
How to confirm that all outgoing traffic is using defined TOR proxy

### DISCUSSION THREAD

**Discussion by andrekrit** [searx/searx#2968] [Installation issue: Docker] [answered]
On my searx instance running on Docker, I've configured a TOR proxy (also running on Docker) as follows:
```
    proxies:
        all:// : socks5://172.17.0.5:9150
```
I've confirmed that the searx container does have an open connection to the the proxy instance, but I'm unable to tell or confirm that traffic is indeed routed via the proxy only. In fact, I've stopped the proxy container and I still get search results! Have I misconfigured or missed something and is there a way to actually confirm that the proxy is being used?

Another question on the same topic: what does `using_tor_proxy : True` do? How is it related to the defined proxies?

**Comment by MarcAbonce** [comment, marked as answer]
Thank you for noticing this issue!
I just opened https://github.com/searx/searx/pull/2969 that should hopefully fix it.

By the way, if you're using Tor you should be using `socks5h` with the **h** at the end, otherwise the DNS lookups are run in your local, see: https://curl.se/libcurl/c/CURLOPT_PROXY.html

The `using_tor_proxy` setting is actually just to enable a couple of .onion engines that are otherwise disabled. Proxying through Tor should still work fine regardless. Now I realise that the setting's name is probably a bit confusing.
